frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,18 @@
+import { Navigate } from '~/libs/components/components.js';
+import { AppRoute } from '~/libs/enums/enums.js';
+import { useAppSelector } from '~/libs/hooks/hooks.js';
+
+const ProtectedRoute: React.FC<React.PropsWithChildren> = ({
```

Comment:
```suggestion
const ProtectedRoute: React.FC<Properties> = ({
```
create your type and describe children there